### PR TITLE
docs: Add ASF attribution

### LINKED ISSUE #950: Use correct attribution in footer of documentation pages
### Is your feature request related to a problem or challenge?

To be inline with ASF policy, we should update the text in the footer used in the docs  to match the footer used in the main website (https://arrow.apache.org/). i.e. add the trademark attribution test.

I am not sure where the docs are hosted 

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by simicd**
# Which issue does this PR close?
Closes #950 

 # Rationale for this change
Aligning Apache Software Foundation attribution across projects

# What changes are included in this PR?
- Add ASF footer
- Update docs requirements.txt (align with arrow-datafusion)
- Fix issues preventing docs compilation

# Are there any user-facing changes?
Updated footer on documentation website

image

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -31,13 +31,11 @@
...
 # sys.path.insert(0, os.path.abspath('.'))
 
-import datafusion
-
 # -- Project information -----------------------------------------------------
 
-project = 'Arrow DataFusion'
-copyright = '2022, Apache Software Foundation'
-author = 'Arrow DataFusion Authors'
+project = 'Apache Arrow DataFusion'
```
Comment (andygrove) [inline docs/source/conf.py:36]:
```suggestion
project = 'Apache Arrow Ballista'
```

File: docs/source/_templates/layout.html [hunk 2]
```diff
@@ -3,3 +3,24 @@
...
+<!-- Based on pydata_sphinx_theme/footer.html -->
+<footer class="footer mt-5 mt-md-0">
+  <div class="container">
+    {% for footer_item in theme_footer_items %}
+    <div class="footer-item">
+      {% include footer_item %}
+    </div>
+    {% endfor %}
+    <div class="footer-item">
+      <p>Apache Arrow DataFusion, Arrow DataFusion, Apache, the Apache feather logo, and the Apache Arrow DataFusion project logo</p>
```
Comment (andygrove) [inline docs/source/_templates/layout.html:20]:
```suggestion
      <p>Apache Arrow Ballista, Arrow Ballista, Apache, the Apache feather logo, and the Apache Arrow Ballista project logo</p>
```

**Comment by simicd** [comment]
Thanks for your feedback @andygrove! Made the updates and all checks are passing

**Review by andygrove — APPROVED** [review]
Thank you @simicd

**Pull request merged**